.github/workflows/docker.yml

```diff
@@ -0,0 +1,49 @@
+﻿name: Docker Upload
+
+on:
+  push:
+    branches:
+      - master
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-upload:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: "6.0.100"
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.repository_owner }}
```

Comment:
Username coming from the repo owner seems non-ideal. Is this meant for internal usage only? Seems like making it an envvar allows for it to change in the future and can still be set to github user for now, but that only makes sense if this is meant to be a more universal image instead of being internal only

Reply:
I can change the username to pull from a envvar. [Should be noted that this was copied from Hive too](https://github.com/Atlas-Rhythm/Hive/blob/master/.github/workflows/docker.yml#L37)